diff -up a/libraries/liblmdb/midl.h b/libraries/liblmdb/midl.h
--- a/libraries/liblmdb/midl.h	2018-03-22 16:23:05.000000000 +0100
+++ b/libraries/liblmdb/midl.h	2024-03-21 11:46:54.616000000 +0100
@@ -56,7 +56,7 @@ typedef MDB_ID *MDB_IDL;
 /* IDL sizes - likely should be even bigger
  *   limiting factors: sizeof(ID), thread stack size
  */
-#define	MDB_IDL_LOGN	16	/* DB_SIZE is 2^16, UM_SIZE is 2^17 */
+#define	MDB_IDL_LOGN	18	/* DB_SIZE is 2^16, UM_SIZE is 2^17 */
 #define MDB_IDL_DB_SIZE		(1<<MDB_IDL_LOGN)
 #define MDB_IDL_UM_SIZE		(1<<(MDB_IDL_LOGN+1))

