--- a/bundled/gcp/google-cloud-sdk/bin/gcloud	1980-01-01 09:00:00.000000000 +0100
+++ b/bundled/gcp/google-cloud-sdk/bin/gcloud	2021-10-14 11:30:17.726138166 +0200
@@ -128,6 +128,11 @@
   fi
 }
 
+if [ -z "$CLOUDSDK_PYTHON" ]; then
+  CLOUDSDK_PYTHON="/usr/libexec/platform-python"
+  CLOUDSDK_PYTHON_SITEPACKAGES=1
+fi
+
 setup_cloudsdk_python
 
 # $PYTHONHOME can interfere with gcloud. Users should use
